src/storage/src/hummock/compactor/iterator.rs

```diff
@@ -361,49 +365,28 @@ impl ConcatSstableIterator {
                 .sstable(table_info, &mut self.stats)
                 .verbose_instrument_await("stream_iter_sstable")
                 .await?;
-            let block_metas = &sstable.meta.block_metas;
-            let mut start_index = match seek_key {
-                None => 0,
+
+            let filter_key_range = match seek_key {
                 Some(seek_key) => {
-                    // start_index points to the greatest block whose smallest_key <= seek_key.
-                    block_metas
-                        .partition_point(|block| {
-                            seek_key.cmp(&FullKey::decode(&block.smallest_key)) != Ordering::Less
-                        })
-                        .saturating_sub(1)
+                    KeyRange::new(seek_key.encode().into(), self.key_range.right.clone())
```

Comment:
What happen if `seek_key > self.key_range.right` or `seek_key < self.key_range.left`? It seems that we didn't handle the corner cases when `seek_key` is out of range here.

Reply:
How about covering these cases with unit tests?

Reply:
UT has cover these two case

And we have handle the `seek_key` and `key_range` in `fn seek` 

```rust
/// Resets the iterator and seeks to the first position where the stored key >= `key`.
    async fn seek<'a>(&'a mut self, key: FullKey<&'a [u8]>) -> HummockResult<()> {
        let seek_key = if self.key_range.left.is_empty() {
            key
        } else {
            match key.cmp(&FullKey::decode(&self.key_range.left)) {
                Ordering::Less | Ordering::Equal => FullKey::decode(&self.key_range.left),
                Ordering::Greater => key,
            }
        };
        let table_idx = self.sstables.partition_point(|table| {
            // We use the maximum key of an SST for the search. That way, we guarantee that the
            // resulting SST contains either that key or the next-larger KV-pair. Subsequently,
            // we avoid calling `seek_idx()` twice if the determined SST does not contain `key`.

            // Note that we need to use `<` instead of `<=` to ensure that all keys in an SST
            // (including its max. key) produce the same search result.
            let max_sst_key = &table.key_range.as_ref().unwrap().right;
            FullKey::decode(max_sst_key).cmp(&seek_key) == Ordering::Less
        });

        self.seek_idx(table_idx, Some(key)).await
    }
```

Reply:
It can be `KeyRange::new(max(seek_key.encode().into(), self.key_range.left if not None), self.key_range.right.clone())`